bundles/io/org.eclipse.smarthome.io.transport.mqtt/src/main/java/org/eclipse/smarthome/io/transport/mqtt/MqttService.java

```diff
@@ -25,215 +31,375 @@
  * message publishers and subscribers. This service is the main entry point for all bundles wanting to use the MQTT
  * transport.
  *
+ * @author David Graeff - Added/Removed observer interface, Add/Remove/Enumerate broker connections.
  * @author Davy Vanherbergen
  * @author Markus Rathgeb - Synchronize access to broker connections
  */
 public class MqttService implements ManagedService {
-
+    private static final String NAME_PROPERTY = "name";
     private final Logger logger = LoggerFactory.getLogger(MqttService.class);
-
     private final Map<String, MqttBrokerConnection> brokerConnections = new ConcurrentHashMap<String, MqttBrokerConnection>();
-
+    private final List<MqttBrokersObserver> brokersObservers = new CopyOnWriteArrayList<>();
+    @Deprecated
     private EventPublisher eventPublisher;
 
-    @Override
-    public void updated(Dictionary<String, ?> properties) throws ConfigurationException {
-
-        // load broker configurations from configuration file
-        if (properties == null || properties.isEmpty()) {
-            return;
-        }
-
+    /**
+     * Transform the JDK 1.0 old-school, rotten, deprecated Dictionary content to something useful.
+     *
+     * @param properties An old-school, deprecated dictionary object with the service configuration.
+     * @return A 'list' of broker configurations as key-value maps. A configuration map at least contains a "name".
+     */
+    public Map<String, Map<String, Object>> extractBrokerConfigurations(Dictionary<String, ?> properties) {
+        Map<String, Map<String, Object>> configPerBroker = new TreeMap<String, Map<String, Object>>();
+        // We can't enumerate all key-value pairs, only the keys and then we need to perform a lookup
+        // for each value. Inefficient, but can't be solved differently for now.
         Enumeration<String> keys = properties.keys();
         while (keys.hasMoreElements()) {
-
             String key = keys.nextElement();
-
+            String value = (String) properties.get(key);
+            // ignore the only non-broker property..
             if (key.equals("service.pid")) {
-                // ignore the only non-broker property..
                 continue;
             }
 
             String[] subkeys = key.split("\\.");
-            if (subkeys.length != 2) {
-                logger.debug("MQTT Broker property '{}' should have the format 'broker.propertykey'", key);
+            if (subkeys.length != 2 || StringUtils.isBlank(value)) {
```

Comment:
No they are not a problem therefore we don't fail here, but log the unexpected name.key=no_value triple as debug log level. Improper used "key=" lines should be commented out instead of keeping the parser busy, I reckon.